framework/src/base/MaxQpsThread.C

```diff
@@ -63,26 +65,26 @@ MaxQpsThread::operator() (const ConstElemRange & range)
       // We cannot mess with the FE objects in Assembly, because we might need to request second derivatives
       // later on. If we used them, we'd call reinit on them, thus making the call to request second
       // derivatives harmful (i.e. leading to segfaults/asserts). Thus, we have to use a locally allocated object here.
-      FEBase * fe = FEBase::build(dim, fe_type).release();
+      UniquePtr<FEBase> fe (FEBase::build(dim, fe_type));
 
       // figure out the number of qps for volume
-      QBase * qrule = QBase::build(_qtype, dim, _order).release();
-      fe->attach_quadrature_rule(qrule);
+      UniquePtr<QBase> qrule (QBase::build(_qtype, dim, _order));
+      fe->attach_quadrature_rule(qrule.get());
```

Comment:
I just think this kind of thing is incredibly unnecessary.  This is created, used and destroyed all within the same method.  

Not only that... but you go through the trouble of creating a `UniquePtr` (whatever the hell that is) only to turn around and call `get()` on it to turn it back into a regular pointer...

I'm not talking about optimization... just code clarity and... why?

Are pointers forbidden technology now?

I can see a use-case (maybe) for the shared pointer stuff in other parts of MOOSE... where we need to create objects in one place that are held in another place (like with the factories).  There it kind-of makes sense.  But I don't see any reason to remove all pointer usage from a C++ library!